### PR TITLE
feat(retry): Add retry logic

### PR DESCRIPTION
**Pull request by lvalerom**
If a call to any of the providers' APIs fails we do not retry and the entire run fails. This PR adds a retry logic to retry for a few minutes before failing the entire run.

### CODE REVIEW

File: pkg/crawlers/azure/azure.go [hunk 1]
```diff
@@ -278,12 +277,20 @@ func (c *azureNetworkCrawler) fetchAll() ([][]byte, error) {
...
+			}
+			if jsonURL == "" {
+				return errors.Errorf("failed to crawl Azure with URL: %s, empty JSON URL returned. This could indicate Azure's services are unavailable", url)
+			}
+			return nil
+		})
+		if retryErr != nil {
+			return nil, retryErr
 		}
 		log.Printf("Received Azure network JSON URL: %s", jsonURL)
```
Comment (vikin91) [inline pkg/crawlers/azure/azure.go:295]: I know, this is not part of this PR, but in case there is another push planned, let's improve this message.

 ```suggestion
		log.Printf("Success obtaining Azure network JSON URL %q from %q", jsonURL, url)
```
(sorry, not sure how those 3 spaces got there into the diff - I somehow cannot remove it from the level of review comment)

File: pkg/crawlers/azure/azure.go [hunk 2]
```diff
@@ -318,7 +325,7 @@ func (c *azureNetworkCrawler) redirectToJSONURL(rawURL string) (string, error) {
 		rawURL)
 	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
 	if err != nil {
-		errors.Wrapf(err, "failed to redirect to JSON URL while trying to crawl Azure with URL: %s", rawURL)
+		err = errors.Wrapf(err, "failed to redirect to JSON URL while trying to crawl Azure with URL: %s", rawURL)
```
Comment (vikin91) [inline pkg/crawlers/azure/azure.go:328]:
```suggestion
		err = errors.Wrapf(err, "failed to redirect to JSON URL %q while trying to crawl Azure with URL", rawURL)
```

**Review by vikin91 — APPROVED** [review]
LGTM! By looking at https://github.com/stackrox/external-network-pusher/actions/runs/12264605728?pr=146 I see that it was tested and works.

I started commenting on potential further improvements, but I saw there are much more improvements possible than those two, so feel free to skip those.

File: pkg/crawlers/azure/azure.go (hunk 1 above)
Comment (lvalerom) [reply pkg/crawlers/azure/azure.go]: Done

File: pkg/crawlers/azure/azure.go (hunk 2 above)
Comment (lvalerom) [reply pkg/crawlers/azure/azure.go]: Done

**Comment by lvalerom** [comment]
> I started commenting on potential further improvements, but I saw there are much more improvements possible than those two, so feel free to skip those

Yeah there are a few inconsistencies, especially in the log messages. Maybe we can open a follow-up PR at some point.
